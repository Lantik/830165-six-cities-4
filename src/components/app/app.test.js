import React from 'react';
import renderer from 'react-test-renderer';
import App from './app.jsx';

const mock = {
  offers: [
    {
      id: 1,
      price: 30,
      rating: 43,
      name: `Beautiful & luxurious apartment at great location`,
      src: `img/apartment-01.jpg`,
      type: `Apartment`,
      inBookmarks: true
    },
    {
      id: 2,
      price: 54,
      rating: 67,
      name: `Wood and stone place`,
      src: `img/apartment-02.jpg`,
      type: `Apartment`,
      inBookmarks: true
    },
    {
      id: 3,
      price: 400,
      rating: 100,
      name: `Stone and wood place`,
      src: `img/apartment-03.jpg`,
      type: `Apartment`,
      inBookmarks: true
    },
    {
      id: 4,
      price: 10,
      rating: 70,
      name: `Luxurious & Beautiful apartment at great location`,
      src: `img/apartment-small-03.jpg`,
      type: `Apartment`,
      inBookmarks: false
    }
  ]
};

it(`<App/> rendered correctly`, () => {
  const {offers} = mock;
  const tree = renderer
    .create(<App
      offers={offers}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
