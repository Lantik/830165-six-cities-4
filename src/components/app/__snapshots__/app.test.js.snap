// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> rendered correctly 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      exact={true}
      path="/"
    >
      <Main
        activeCity="Test city"
        offers={
          Array [
            Object {
              "city": "Amsterdam",
            },
            Object {
              "city": "Capenhagen",
            },
            Object {
              "city": "Test city",
            },
          ]
        }
        onCityTitleClick={[Function]}
        onOfferHeaderClick={[Function]}
        onSortOptionClick={[Function]}
        sortType=""
      />
    </Route>
    <Route
      exact={true}
      path="/offer"
    >
      <OfferDetails
        offer={
          Object {
            "city": "Amsterdam",
          }
        }
        offers={
          Array [
            Object {
              "city": "Amsterdam",
            },
            Object {
              "city": "Capenhagen",
            },
            Object {
              "city": "Test city",
            },
          ]
        }
        onOfferHeaderClick={[Function]}
      />
    </Route>
    <Route
      exact={true}
      path="/dev-component"
    />
  </Switch>
</BrowserRouter>
`;
